src/script/components/MessagesList/Message/MemberMessage/MessageContent.tsx

```diff
@@ -17,10 +17,193 @@
+      }
+
+      if (message.user().isMe) {
+        return exceedsMaxVisibleUsers
+          ? t('conversationMemberJoinedYouMore', {count: hiddenUsersCount.toString(), users: accusativeUsers})
+          : t('conversationMemberJoinedYou', accusativeUsers);
+      }
+      return exceedsMaxVisibleUsers
+        ? t('conversationMemberJoinedMore', {count: hiddenUsersCount.toString(), name, users: accusativeUsers})
+        : t('conversationMemberJoined', {name, users: accusativeUsers});
+    }
+
+    case CONVERSATION_EVENT.MEMBER_LEAVE: {
+      if (message.reason === MemberLeaveReason.LEGAL_HOLD_POLICY_CONFLICT) {
+        const replaceLinkLegalHold = replaceLink(
+          Config.getConfig().URL.SUPPORT.LEGAL_HOLD_BLOCK,
+          '',
+          'read-more-legal-hold',
+        );
+        if (message.userEntities().some(user => user.isMe)) {
+          return t('conversationYouRemovedMissingLegalHoldConsent', {}, replaceLinkLegalHold);
+        }
+        const users = generateNames(targetedUsers);
+
+        if (message.userEntities().length === 1) {
+          return t('conversationMemberRemovedMissingLegalHoldConsent', users, replaceLinkLegalHold);
+        }
+        if (exceedsMaxVisibleUsers) {
+          return t(
+            'conversationMultipleMembersRemovedMissingLegalHoldConsentMore',
+            {
+              count: hiddenUsersCount.toString(10),
+              users,
+            },
+            replaceLinkLegalHold,
+          );
+        }
+        return t('conversationMultipleMembersRemovedMissingLegalHoldConsent', users, replaceLinkLegalHold);
+      }
+      const temporaryGuestRemoval = message.otherUser().isMe && message.otherUser().isTemporaryGuest();
+      if (temporaryGuestRemoval) {
+        return t('temporaryGuestLeaveMessage');
+      }
+
+      const senderLeft = matchQualifiedIds(message.otherUser(), actor);
+      if (senderLeft) {
+        return message.user().isMe ? t('conversationMemberLeftYou') : t('conversationMemberLeft', name);
+      }
+
+      const allUsers = generateNames(targetedUsers);
+      if (!actor.id) {
+        return t('conversationMemberWereRemoved', allUsers);
+      }
+      return actor.isMe
+        ? t('conversationMemberRemovedYou', allUsers)
+        : t('conversationMemberRemoved', {name, users: allUsers});
+    }
+
+    case ClientEvent.CONVERSATION.TEAM_MEMBER_LEAVE: {
+      return t('conversationTeamLeft', name);
+    }
+
+    default:
+      break;
+  }
+  return '';
+}
 
-export function MessageContent({message}: {message: MemberMessageEntity}) {
-  const {htmlCaption} = useKoSubscribableChildren(message, ['htmlCaption']);
-  return <p className="message-header-caption" dangerouslySetInnerHTML={{__html: htmlCaption}}></p>;
+export function MessageContent({message, onClickAction}: {message: MemberMessageEntity; onClickAction: () => void}) {
+  const htmlCaption = getContent(message);
+  const content = replaceReactComponents(htmlCaption, [
```

Comment:
This is where the magic happens. We input a string like `"hello <strong>felix</strong>"` and, with the right config, the `<strong>` part will be replace with a react component.